src/main/frontend/src/components/form/Button.svelte

```diff
@@ -9,28 +13,24 @@
 	export let href = "";
 	export let disabled = false;
 
-	let btnClass = outlined ? color+"-color " + color+"-text" : color+"-color ";
-</script>
+	function clicked() {
+		if (href) {
+			goto(href);
+			return;
+		} else {
+			dispatch("click");
+		}
+	}
 
+	$: btnClass = outlined ? color+"-color " + color+"-text" : disabled ? disabled : color+"-color ";
+
+</script>
 
-{#if !href}
-<MatButton class={"m-1 " + btnClass} {...$$restProps} disabled={disabled} on:click {outlined}>
+<MatButton class={"m-1 " + btnClass} {...$$restProps} disabled={disabled} on:click={clicked} {outlined}>
 	<span>
 		{#if icon}
 			<Icon path={icon}/>
 		{/if}
 		{text}
 	</span>
-</MatButton>
-{:else}
-	<a href={href} rel="noopener">
```

Comment:
Just so I know: Why not do it this way? Is goto(href) doing something different?

Reply:
MatButton was repetitive, goto is sapper built-in function, so it's generally good to use the framework functions

Reply:
Oh I completely overlooked it being repetitive. Makes sense.